data/basics/userBalance.go

```diff
@@ -181,12 +270,35 @@ type BalanceDetail struct {
 // up the creator of the asset, whose balance record contains the AssetParams
 type AssetIndex uint64
 
-// AssetLocator stores both the asset creator, whose balance record contains
-// the asset parameters, and the asset index, which is the key into those
-// parameters
-type AssetLocator struct {
+// AppIndex is the unique integer index of an application that can be used to
+// look up the creator of the application, whose balance record contains the
+// AppParams
+type AppIndex uint64
+
+// CreatableIndex represents either an AssetIndex or AppIndex, which come from
+// the same namespace of indices as each other (both assets and apps are
+// "creatables")
+type CreatableIndex uint64
+
+// CreatableType is an enum representing whether or not a given creatable is an
+// application or an asset
+type CreatableType uint64
+
+const (
+	// AssetCreatable is the CreatableType corresponding to assets
+	AssetCreatable CreatableType = 0
```

Comment:
Consider using `iota` here

Reply:
Ok, I'm curious about your opinion on this -- in this case when I upgrade the database I'm doing `ADD COLUMN ctype INTEGER DEFAULT 0` to the table containing asset IDs/app IDs. The table currently only contains assets. So I really need the `AssetCreatable` `CreatableType` to be 0, and not just distinct from the `AppCreatable` type. Would you still use `iota` in this case?

Reply:
Regardless of whether you choose to use `iota` or just zero, it would be a good idea to document the reason you've just mentioned.

Reply:
For sure, added a comment, thanks :+1: